File: Server/src/main/java/com/codingfairy/bl/service/VisitCircleService.java
package com.codingfairy.bl.service;

import com.codingfairy.bl.vo.VisitCircleVo;
import com.codingfairy.exception.ParamException;
import com.codingfairy.utils.enums.QueryThreshold;
import com.codingfairy.web.json.Tuple;
import com.codingfairy.web.json.analysis.element.VisitingCycleElement;

import java.util.List;

/**
 * Created by cuihao on 2017-05-16.
 * visit circle service
 */
public interface VisitCircleService {

    List<VisitCircleVo> getVisitCircleStat(int startHour, QueryThreshold threshold, int startRange, String probeId) throws ParamException;

    List<Tuple<String,Number>> findByHourAndProbe(int hour, String probeId);

    VisitCircleVo findById(int id);

    VisitCircleVo save(VisitingCycleElement visitingCycleElement);
}

File: StatDisplay/src/services/visitCircleService.js
/**
 * Created by yyy on 2017/5/27.
 */
import {request} from "../utils/request";

export async function getVisitCircle(params) {
  return request(`/visitCircle/stat`,{
    method: `POST`,
    body: JSON.stringify({
      probeId: params.probeId,
      startHour: params.startHour,
      startRange: params.startRange,
      threshold: params.threshold
    })
  })
}

export async function getVisitCircleDetail(params) {
  return request(`/visitCircle/detail`,{
    method: `POST`,
    body: JSON.stringify({
      hour: params.hour,
      probeId: params.probeId
    })
  })
}
